Polyhedron/demo/Polyhedron/Viewer.cpp

```diff
@@ -1090,317 +1092,87 @@ void Viewer::resizeGL(int w, int h)
-        {
-            std::cerr<<"adding vertex shader FAILED"<<std::endl;
-        }
-        if(!program->addShaderFromSourceFile(QOpenGLShader::Fragment,":/cgal/Polyhedron_3/resources/shader_c3t3.f" ))
-        {
-            std::cerr<<"adding fragment shader FAILED"<<std::endl;
-        }
-        program->bindAttributeLocation("colors", 1);
-        program->link();
-        d->shader_programs[PROGRAM_CUTPLANE_SPHERES] = program;
-        return program;
-      }
+      return declare_program(name, ":/cgal/Polyhedron_3/resources/shader_c3t3_spheres.v" , ":/cgal/Polyhedron_3/resources/shader_c3t3.f");
+     break;
     case PROGRAM_C3T3_TETS:
-      if( d->shader_programs[PROGRAM_C3T3_TETS])
-      {
-          return d->shader_programs[PROGRAM_C3T3_TETS];
-      }
-      else
-      {
-        QOpenGLShaderProgram *program = new QOpenGLShaderProgram(viewer);
-        if(!program->addShaderFromSourceFile(QOpenGLShader::Vertex,":/cgal/Polyhedron_3/resources/shader_c3t3_tets.v"))
-        {
-            std::cerr<<"adding vertex shader FAILED"<<std::endl;
-        }
-        if(!program->addShaderFromSourceFile(QOpenGLShader::Fragment,":/cgal/Polyhedron_3/resources/shader_with_light.f" ))
-        {
-            std::cerr<<"adding fragment shader FAILED"<<std::endl;
-        }
-        program->bindAttributeLocation("colors", 1);
-        program->link();
-        d->shader_programs[PROGRAM_C3T3_TETS] = program;
-        return program;
-      }
+      return declare_program(name, ":/cgal/Polyhedron_3/resources/shader_c3t3_tets.v" , ":/cgal/Polyhedron_3/resources/shader_with_light.f");
+     break;
     case PROGRAM_SPHERES:
-        if( d->shader_programs[PROGRAM_SPHERES])
-        {
-            return d->shader_programs[PROGRAM_SPHERES];
-        }
-        else
-        {
-            QOpenGLShaderProgram *program = new QOpenGLShaderProgram(viewer);
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Vertex,":/cgal/Polyhedron_3/resources/shader_spheres.v" ))
-            {
-                std::cerr<<"adding vertex shader FAILED"<<std::endl;
-            }
-            if(!program->addShaderFromSourceFile(QOpenGLShader::Fragment,":/cgal/Polyhedron_3/resources/shader_with_light.f" ))
-            {
-                std::cerr<<"adding fragment shader FAILED"<<std::endl;
-            }
-            program->bindAttributeLocation("colors", 1);
-            program->link();
-            d->shader_programs[PROGRAM_SPHERES] = program;
-            return program;
-
-        }
+      return declare_program(name, ":/cgal/Polyhedron_3/resources/shader_spheres.v" , ":/cgal/Polyhedron_3/resources/shader_with_light.f");
+      break;
+    case PROGRAM_FLAT:
+      return declare_program(name, ":/cgal/Polyhedron_3/resources/shader_with_light.v", ":/cgal/Polyhedron_3/resources/shader_flat.f");
       break;
 
     default:
         std::cerr<<"ERROR : Program not found."<<std::endl;
         return 0;
     }
-}
```

Comment:
This cannot compile. There is a `}` missing. Please test your code, and push the fix.